docs/DeveloperGuide.md

```diff
@@ -266,31 +266,196 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to list all tasks.
+2.  GrAB3 shows a list of tasks from the TaskSchedule.
+3.  User requests to delete a specific task in the TaskSchedule.
+4.  GrAB3 deletes the task.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 2a. The TaskSchedule is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. The specified task does not exist.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. GrAB3 shows an error message.
 
       Use case resumes at step 2.
 
+**UC03: Edit a task**
+
+**Preconditions: Task exists in TaskSchedule**
+
+**Guarantees: Task specified by user gets edited**
+
+**MSS**
+
+1.  User requests to list all tasks.
+2.  GrAB3 shows a list of tasks from TaskSchedule.
+3.  User requests to edit a specific task in TaskSchedule.
+4.  GrAB3 edits the description of the task.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The TaskSchedule is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. GrAB3 shows an error message.
+
+    Use case resumes at step 2.
+    
+**UC04: View TaskTemplateList**
+
+**Preconditions: TaskTemplates in TaskTemplateList exists**
+
+**Guarantees: List of all TaskTemplates available**
+
+**MSS**
+
+1. User request to list TaskTemplateList.
+2. GrAB3 shows a list of TaskTemplates.
+Use case ends
+
+**UC05: Load TaskTemplate**
+
+**Preconditions: Tasks exist in TaskTemplate**
+
+**Guarantees: TaskSchedule with tasks from the TaskTemplate get added for the user**
+
+**MSS**
+1. User requests to view TaskTemplateList (UC04)
+2. Program shows a list of TaskTemplate 
+3. User selects TaskTemplate to load into TaskSchedule
+4. Program adds all tasks from selected TaskTemplate into TaskSchedule
+Use case ends
+
+### Module Schedule
+
+**UC06: Add a module**
+
+**Preconditions: ModuleList must exist**
+
+**Guarantees: A new module will be added to MyModules**
+
+**MSS**
+
+1.  User requests to add a new module into MyModules
+2.  GrAB3 adds the selected module to MyModules
+
+    Use case ends.
+
+**Extensions**
+
+**UC07: Remove a module**
+
+**Preconditions: Selected module must exist in MyModules** 
+
+**Guarantees: The module specified by the user will be removed from MyModules and ModuleSchedule**
+
+**MSS**
+
+1.  User requests to list all modules
+2.  GrAB3 shows a list of modules
+3.  User requests to remove a specific module from ModuleSchedule
+4.  GrAB3 remove the module from MyModules and ModelSchedule
+
+    Use case ends.
+
+**Extensions**
+
+* 3a. The given module code is invalid.
+
+    * 3a1. GrAB3 shows an error message.
+
+    Use case resumes at step 2.
+
+
+**UC08: View ModuleList**
+
+**Preconditions: Modules exist in database**
+
+**Guarantees: List of all modules available**
+
+**MSS**
+
+1. User request to list all modules
+2. GrAB3 shows a list of all modules
+Use case ends
+
+**UC09: View MyModules**
```

Comment:
I think we are going to omit MyModules from the architecture, we will be using ModuleSchedule instead, so this use case should be _View ModuleSchedule_ instead